### PR TITLE
Modeler 3d layout Improvements on primitives handling.

### PR DESCRIPTION
**Pull request by maxcapodi78**
Implemented:
 #2136 
#2135 
#2134

### CODE REVIEW

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

File: pyaedt/modeler/Primitives3DLayout.py [hunk 1]
```diff
@@ -232,6 +236,156 @@ def objects_by_net(self, net_name, object_filter=None, include_voids=False):
             objs = self.modeler.oeditor.FindObjects("Net", net_name)
         return objs
 
+    @pyaedt_function_handler()
+    def _get_names(self, categories):
+        names = []
+        for object in categories:
+            names.extend(self.modeler.oeditor.FindObjects("Type", object))
```
Comment (MaxJPRey) [inline pyaedt/modeler/Primitives3DLayout.py:243]: This should fix the codacy issue.
```suggestion
        for category in categories:
            names.extend(self.modeler.oeditor.FindObjects("Type", category))
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 2]
```diff
@@ -232,6 +236,156 @@ def objects_by_net(self, net_name, object_filter=None, include_voids=False):
...
+
+        Returns
+        -------
+        list
+        """
+        return self._get_names(["poly", "plg"])
+
+    @property
+    def polygon_voids_names(self):
+        """Get the list of all polygons in layout.
```
Comment (MaxJPRey) [inline pyaedt/modeler/Primitives3DLayout.py:258]:
```suggestion
        """Get the list of all void polygons in layout.
```

File: pyaedt/modeler/Model3DLayout.py [hunk 3]
```diff
@@ -564,16 +566,52 @@ def subtract(self, blank, tool):
...
+        -------
+        str or list
+           String or list of the selections.
+
+        """
+
+        if not isinstance(objtosplit, list):
+            objtosplit = [objtosplit]
+        objnames = []
+        for el in objtosplit:
```
Comment (MaxJPRey) [inline pyaedt/modeler/Model3DLayout.py:604]:
```suggestion
    def convert_to_selections(self, objects_to_split, return_list=False):
        """Convert one or more object to selections.

        Parameters
        ----------
        objects_to_split : str, int, list
            One or more objects to convert to selections. A list can contain
            both strings (object names) and integers (object IDs).
        return_list : bool, option
            Whether to return a list of the selections. The default is
            ``False``, in which case a string of the selections is returned.
            If ``True``, a list of the selections is returned.

        Returns
        -------
        str or list
           String or list of the selections.

        """

        if not isinstance(objects_to_split, list):
            objects_to_split = [objects_to_split]
        objnames = []
        for el in objects_to_split:
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Pull request merged**